source/extensions/filters/http/oauth2/filter.cc

```diff
@@ -93,6 +93,26 @@ authScopesList(const Protobuf::RepeatedPtrField<std::string>& auth_scopes_protos
   return scopes;
 }
 
+// Transforms the proto list into encoded resource params
+// Takes care of percentage encoding http and https is needed
+std::string encodeResourceList(const Protobuf::RepeatedPtrField<std::string>& resources_protos) {
+  std::string result = "";
+
+  if (!resources_protos.empty()) {
+    std::string h1 = "http://";
+    std::string h2 = "https://";
+
+    for (const auto& resource : resources_protos) {
+      if (resource.rfind(h1, 0) == 0 || resource.rfind(h2, 0) == 0) {
```

Comment:
Looking at the RFC it seems like the resource must be an absolute URL, so why do we need this special handling? Is there a reason why we would not percent encode in all cases?

cc @williamsfu99 in case you can shed some light on the the spec here

Reply:
resource may be in ARN format hence the special case. 
"resource":"arn:aws:iam::123456789012:user/12345"

https://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html

Reply:
@snowp @williamsfu99 Let me know what do you think about ARN resource. Also code is up for review again

Reply:
I don't have a problem supporting the ARN resources, but it seems like we should percent encode it regardless in order to pass it as query param? Are there use cases in the wild where it expects these query parameters to not be URL encoded?

Reply:
Hi @snowp , here is some information around URI and I agree that it should be url encoded. 

absolute URI: The RFC does not mandate the format of the resource param. It's just an identifier and meant to identify the protected resource the client is requesting a token for. 
` The "resource" parameter URI value is an identifier
      representing the identity of the resource, which MAY be a locator
      that corresponds to a network-addressable location where the
      target resource is hosted. `
 definition of URI: https://tools.ietf.org/html/rfc3986#section-4.3 
 
 I guess, you are in agreement with this. 
 
 I will update the PR with URL encoding next.

Reply:
@snowp its up for review